### ISSUE TITLE
OpenSSL binaries from Indy is captcha protected.

### ISSUE THREAD

**Issue by AlexKotik** [LongDirtyAnimAlf/fpcupdeluxe#98]
fpcupdeluxe hangs when trying to download https://indy.fulgan.com/SSL/openssl-1.0.2q-x64_86-win64.zip I think because the url is protected with a captcha. How do I make fpcupdeluxe work?

**Comment by sstvmaster** [comment]
hmm, i can download using this link.

**Comment by AlexKotik** [comment]
![screenshot_20190211-225142](https://user-images.githubusercontent.com/24991384/52589438-dc9bc180-2e4f-11e9-8434-a0833668baaa.png)

**Comment by sstvmaster** [comment]
Hi, i have tested with Firefox, Chrome (Windows), Chrome, Firefox on Android via LTE and it works. I have a guess, but cant test this.

**Comment by LongDirtyAnimAlf** [comment]
Strange error that I cannot reproduce on my system !
And, if I cannot reproduce, it is very hard to investigate.

Workaround:
Just download these files by yourself and add them into the directory where fpcupdeluxe resides.

**Comment by mdbs99** [comment]
I see the captcha - I believe this is new.
Maybe fpcupdeluxe could put those files in Git or tell users to download manually.

**Comment by sstvmaster** [comment]
I think it's your IP address. You can test it with an VPN service in europe, or use the Opera Browser.

![opera_vpn](https://user-images.githubusercontent.com/19416108/52664684-f568ad80-2f09-11e9-89e4-fa8a443cce12.JPG)

**Comment by mdbs99** [comment]
But fpcupdeluxe is used around the world. I guess it's important to be sure that it will work whatever the IP origin.

**Comment by eugeneloza** [comment]
For me 1.6.2o also hanging up in Windows 7x64 with Ukrainian IP trying to download
```
fpcupdeluxe: WARNING: FPCNativeInstaller (DownloadOpenSSL): No OpenSLL library files available for SSL. Going to download them.
fpcupdeluxe: info: FPCNativeInstaller (GetFile): Downloading https://indy.fulgan.com/SSL/openssl-1.0.2q-x64_86-win64.zip
```
I can download the file through a direct link without capcha or anything else, thou.

> Just download these files by yourself and add them into the directory where fpcupdeluxe resides

Unfortunately it didn't work for me. Placing `openssl-1.0.2q-x64_86-win64.zip` at `fpcupdeluxe-x86_64-win64.exe` or in installation folder still trigger download and hang-up.

I can't download those in earlier versions too. 1.6.2h is a bit more talkative:
```
fpcupdeluxe: WARNING: FPCNativeInstaller (DownloadOpenSSL): No OpenSLL library files available for SSL. Going to download them
fpcupdeluxe: info: FPCNativeInstaller (GetFile): Downloading https://indy.fulgan.com/SSL/openssl-1.0.2q-x64_86-win64.zip
fpcupdeluxe: info: Native downloader: Getting openssl-1.0.2q-x64_86-win64.zip from https://indy.fulgan.com/SSL/
fpcupdeluxe: info: Native downloader: Getting openssl-1.0.2q-x64_86-win64.zip from https://indy.fulgan.com/SSL/
fpcupdeluxe: info: FPCNativeInstaller (GetFile): Could not download file with URL https://indy.fulgan.com/SSL/openssl-1.0.2q-x64_86-win64.zip into C:\Users\User\AppData\Local\Temp (filename: FPCUPTMP00003.zip)
fpcupdeluxe: info: FPCNativeInstaller (GetFile): Downloading https://indy.fulgan.com/SSL/openssl-1.0.2q-x64_86-win64.zip
fpcupdeluxe: info: Native downloader: Getting openssl-1.0.2q-x64_86-win64.zip from https://indy.fulgan.com/SSL/
fpcupdeluxe: info: Native downloader: Getting openssl-1.0.2q-x64_86-win64.zip from https://indy.fulgan.com/SSL/
fpcupdeluxe: info: FPCNativeInstaller (GetFile): Could not download file with URL https://indy.fulgan.com/SSL/openssl-1.0.2q-x64_86-win64.zip into C:\Users\User\AppData\Local\Temp (filename: FPCUPTMP00003.zip)
```

"Run as administrator" didn't help either.

A few notes:

1. I'm almost sure 1.6.2h worked for me in December/January. I.e. this looks like some new issue at `https://indy.fulgan.com` that block some requests.
2. The Windows 7 x 64 installation is "clean" - I just reinstalled it a few days ago (actually, due to this issue :) because I thought I have messed something up with my registry trying to set up android cross-compilation).

**Comment by eugeneloza** [comment]
UPDATE: version 1.6.1m works flawlessly and downloads everything as expected. Theoretically it's possible that 1.6.2x versions get blocked by firewall, however, 1.6.1m "asks for permission", while 1.6.2x don't. However, that "not asking for permission" may be caused by "trust level: super" automatic setting in the firewall.

**Comment by sstvmaster** [comment]
Hi,

you must unzip "openssl...win64.zip" and the then libea32.dll + ssleay32.dll should be in the same folder as the "fpcupdeluxe-x86_64-win64.exe". In older fpcupdeluxe versions that could be possible to update powershell.

Download the lastest from here: https://docs.microsoft.com/en-us/powershell/scripting/install/installing-windows-powershell?view=powershell-6

You can use version 5.1. for Windows 7 and up,

**Comment by Kashaket** [comment]
It's only for Europe users, especially ukraine and russia, which is famous for its spamming/sniffing services.
I wondered, but, my IP is included in the spammer list, which is prevented me from registration on lazarus forum, and also from downloading lazarus on sourceforge, it's quitely annoying to use vpn every time you need to download something from SF, nevertheless, i didn't see any problems when using fpcupdeluxe

**Comment by LongDirtyAnimAlf** [comment]
Thanks for the info.
Closing this issue.
